File: c#/Simple.cs
using Infinispan.HotRod.Impl;
using Infinispan.HotRod.Config;
using Infinispan.HotRod;
using System;
using System.IO;
using System.Collections.Generic;
using System.Threading;

namespace Infinispan.Tutorial
{
    public class Simple
    {
        static void main() {
            // Create a configuration for a locally-running server
            ConfigurationBuilder builder = new ConfigurationBuilder();
            conf = builder.AddServers("127.0.0.1:11222").Build();
            // Initialize the remote cache manager
            remoteManager = new RemoteCacheManager(conf, new DefaultSerializer());
            // Connect to the server
            remoteManager.Start();
            // Store a value
            cache.Put("key", "value");
            // Retrieve the value and print it out
            Console.WriteLine("key = {0}", cache.Get("key"));
            // Stop the cache manager and release all resources
            remoteManager.Stop();
        }
    }
}

File: c++/src/simple.cpp
#include "infinispan/hotrod/ConfigurationBuilder.h"
#include "infinispan/hotrod/RemoteCacheManager.h"
#include "infinispan/hotrod/RemoteCache.h"
#include "infinispan/hotrod/Version.h"

#include <stdlib.h>
#include <iostream>

using namespace infinispan::hotrod;

int main(int argc, char** argv) {
    // Create a configuration for a locally-running server
    ConfigurationBuilder builder;
    builder.addServer().host(argc > 1 ? argv[1] : "127.0.0.1").port(argc > 2 ? atoi(argv[2]) : 11222);
    // Initialize the remote cache manager
    RemoteCacheManager cacheManager(builder.build(), false);
    // Obtain the remote cache
    RemoteCache<std::string, std::string> cache = cacheManager.getCache<std::string, std::string>();
    // Connect to the server
    cacheManager.start();
    // Store a value
    cache.put("key", "value");
    // Retrieve the value and print it out
    std::cout << "key = " << *cache.get("key") << std::endl;
    // Stop the cache manager and release all resources
    cacheManager.stop();
    return 0;
}
